compiler/codegen.js

```diff
@@ -2086,12 +2086,20 @@ const generateCall = (scope, decl, _global, _name, unusedValue = false) => {
   let args = decl.arguments;
   if (func && args.length < paramCount) {
     // too little args, push undefineds
-    args = args.concat(new Array(paramCount - args.length).fill(DEFAULT_VALUE));
+    if (func.hasRestArgument) {
+      args = args.concat(new Array(paramCount - 1 - args.length).fill(DEFAULT_VALUE));
+    } else {
+      args = args.concat(new Array(paramCount - args.length).fill(DEFAULT_VALUE));
+    }
   }
 
-  if (func && args.length > paramCount) {
```

Comment:
Could we avoid deleting this jank if for now/this PR please?

Reply:
Oops, but out of curiosity, why? Doesn't that eliminate any side effects or things like that?
Regardless, I'll move it below the rest argument stuff to preserve that part's functionality

Reply:
> why?

In my testing before it had a negative test262 impact so I'd rather leave it be for now, but I'll likely delete it soon(tm)